packages/components/src/core/Panel/index.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { StyledDrawer, StyledHeaderComponent } from "./style";
+import { DrawerProps } from "@mui/material";
+import PanelHeader from "./components/PanelHeader";
+import PanelHeaderClose, {
+  PanelHeaderCloseProps,
+} from "./components/PanelHeaderClose";
+
+export interface BasicPanelProps extends Omit<DrawerProps, "variant"> {
+  sdsType?: "basic"; // Discriminator
+  position?: "left" | "right";
+  width?: number | string;
+}
+
+export interface OverlayPanelProps extends Omit<DrawerProps, "variant"> {
+  sdsType?: "overlay"; // Discriminator
+  position?: "left" | "right" | "bottom";
+  width?: number | string;
+  headerComponent?: React.ReactNode;
+  closeButtonOnClick?: PanelHeaderCloseProps["onClick"];
+  closeButtonComponent?: PanelHeaderCloseProps["closeButtonComponent"];
+}
+
+// Discriminated Union
+export type PanelProps = BasicPanelProps | OverlayPanelProps;
+
+// Type guard to narrow the type
+function isOverlayPanelProps(props: PanelProps): props is OverlayPanelProps {
+  return props.sdsType === "overlay";
+}
+
+export const PANEL_BASIC_MIN_WIDTH = 240;
+export const PANEL_OVERLAY_MIN_WIDTH = 320;
+
+/**
+ * @see https://mui.com/material-ui/react-drawer/
+ */
+
+const Panel = React.forwardRef<HTMLDivElement, PanelProps>((props, ref) => {
+  const { children, sdsType = "basic", position = "left", width } = props;
+
+  const DrawerWidth = width
+    ? width
+    : sdsType === "basic"
+      ? PANEL_BASIC_MIN_WIDTH
+      : PANEL_OVERLAY_MIN_WIDTH;
+  const DrawerVariant = sdsType === "basic" ? "persistent" : "temporary";
+
+  // (masoudmanson): The basic Panel only supports "left" or "right" positions.
+  // If a "bottom" position is provided for a basic Panel, it defaults to "left".
+  const DrawerAnchor =
+    sdsType === "overlay"
+      ? position
+      : position === "bottom"
+        ? "left"
+        : position;
+
+  return (
+    <StyledDrawer
+      {...props}
```

Comment:
Hmm, we actually need to override the props that come after `{...props}`, so I thought placing them afterward would ensure our values override any user input. 

For example, we don’t want users to position a `BasicPanel` at the `bottom` of the page, since the SDS design specifies it should only be anchored to the `left` or `right`. Therefore, the `anchor` prop needs to be overridden by our value.

Reply:
Ooh got it! So the spread before is intentional 👌 That makes sense! If so, maybe we can just add a comment to express the intention, in case a future refactor accidentally move it to below lol Thank you!!

Reply:
Absolutely, I'll add a comment to make sure we won't change this in future :D